src/component/ViewInvoice/index.jsx

```diff
@@ -1,6 +1,104 @@
-function ViewInvoice() {
+import { useParams } from "react-router-dom"
+import ViewHeader from "../ViewHeader"
+import { useEffect, useState } from "react"
+import moment from "moment"
+import Bill from "../Bill"
+import StatusIcon from "../StatusIcon"
+
+function ViewInvoice({ }) {
+    const { idName, id } = useParams()
+    const [clientInvoice, setClientInvoice] = useState({
+        name: "",
+        streetAddress: "",
+        city: "",
+        created: "",
+        due: "",
+        details: [],
+        paidToDate: "",
+        invoiceTotal: "",
+        statusName: ""
+    })
+    const { name, streetAddress, city, created, due, details, paidToDate, invoiceTotal, statusName } = clientInvoice
```

Comment:
hmmm

This is like you didn't want to have to do loads of bits of state, but don't want to have to do clientInvoice.name in your jsx.

I'm not sure if I like it or not.

I think I'll allow it, but in future either make separate state for each item, or access the object in the JSX

Reply:
Looking at it more, it seems like in your JSX you're using a mixture of these consts, and access the state directly. I reckon bin this const and just do clientInvoice. in your JSX